src/drive/mobile/modules/mediaBackup/duck/tmp.js

```diff
@@ -0,0 +1,63 @@
+/* global cozy */
+/* 
+This file is TEMPORARY. It covers a stack bug and we need to synchronize again some files
+
+!!TO BE DELETED WHEN datas are OK
+
+*/
+import { updateLibraryItem } from 'drive/mobile/lib/media'
+import { CURRENT_UPLOAD_PROGRESS } from './reducer'
+import throttle from 'lodash/throttle'
+import localforage from 'localforage'
+
+export const checkCorruptedFiles = async (
+  photosOnDevice,
+  alreadyUploaded,
+  client,
+  dispatch
+) => {
+  const onProgressUpdate = progress => {
+    dispatch({ type: CURRENT_UPLOAD_PROGRESS, progress })
+  }
+  const onThumbnailGenerated = thumbnailUrl => {}
+  const seen = (await localforage.getItem('CORRUPTED_FILES')) || false
+  if (seen === false) {
+    const requestFiles = await cozy.client.fetchJSON('GET', '/files/fsck')
+    await localforage.setItem('CORRUPTED_FILES', requestFiles)
+  }
+  const files = await localforage.getItem('CORRUPTED_FILES')
+  const ids = files.reduce(async (memo, value) => {
+    if (value.file_doc && value.file_doc.name) {
+      photosOnDevice.some(async photo => {
+        if (photo.fileName === value.file_doc.name) {
+          try {
+            await updateLibraryItem(
+              value.file_doc,
+              photo,
+              throttle(onProgressUpdate, 500),
+              onThumbnailGenerated
+            )
+            const newObj = files.reduce((acc, v) => {
+              if (v.file_doc._id === value.file_doc._id) {
+                console.log('ici')
+              } else {
+                acc.push(value)
+              }
+              return acc
+            }, [])
+            await localforage.setItem('CORRUPTED_FILES', newObj)
+            return memo.push(photo.id)
+          } catch (error) {
+            console.log('error on CORRUPTED FILES', error)
+          }
+        }
+      })
+    }
+    return memo
+  }, [])
```

Comment:
Maybe we can rewrite the code block above in a more functional style to make things clearer? There's a `reduce` inside a `some` inside a `reduce`, it makes thingsa bit hard to follow along.

Maybe something like:

```
files.filter(onlyFilesWithNames).map(getPhotoOnDevice).forEach(reuploadFile)
```

Do you see what I mean? If not I think at least we should rename some of the variables like `ids`, `value` and `v` so that they express their purpose :)

Reply:
Come on! Nothing more functional than `reduce`. I really don't like chaining, so I'll juste rename some vars as advised :)

Reply:
Well it's a `reduce` call but there are definitely side effects :p. I also think we can use `forEach` instead of `some` if we leave the overall structure as it is, since we're not doing anything with the return value of `some`.

Reply:
Why using `forEach`? We can't break the loop with `forEach`. With `some`, when we return something, it breaks the loop, so I don't iterate over all files again and again.